### ISSUE TITLE
Windows 10 - miner.exe crashing every ~hr

### ISSUE THREAD

**Issue by effofxprime** [develsoftware/GMinerRelease#326]
Hello!  I have been using gminer to mine Cortex since the beginning of this year.  After the latest update, I started to have awesome miner report that it was restarting mining process since it had stopped.  This has been happening frequently and only recently, within the last week maybe two.

Included is a screenshot of the event viewer log report.  I am unsure of where or what else would be helpful in trying to figure out why this has started happening.

Gminer is the latest version.  Windows 10 and fully updated.  Cuda 11.4 Toolkit w/drivers (whatever drivers are included, i don't update from what is included in cuda).
![gminer crashing report](https://user-images.githubusercontent.com/53557/127172577-341fa456-c315-4b20-a408-a488ff6e9efa.png)

**Comment by schaijik** [comment]
What does the mining console say when the crash happens?

**Comment by effofxprime** [comment]
> What does the mining console say when the crash happens?

"Miner crashed, watchdog restarting."
I didn't see an indication as to why, like GPU failed or anything like that.  It was fast and I couldn't screenshot before it closed and restarted.

I only saw this for the first time today because of how automated it runs.  I use awesome miner.  But I turned on the watchdog for it to restart the process without hanging the rig and requiring a restart.
I didn't see anything in the window about a why.  However, I have turned on logging.  Let me check the log, screenshot and get back with that information as well.

**Comment by NamasteIndia** [comment]
> > What does the mining console say when the crash happens?
> 
> "Miner crashed, watchdog restarting."
> I didn't see an indication as to why, like GPU failed or anything like that. It was fast and I couldn't screenshot before it closed and restarted.
> 
> I only saw this for the first time today because of how automated it runs. I use awesome miner. But I turned on the watchdog for it to restart the process without hanging the rig and requiring a restart.
> I didn't see anything in the window about a why. However, I have turned on logging. Let me check the log, screenshot and get back with that information as well.

Did you try disabling all the antivirus features? Try changing UAC to allow all. Reinstall NVIDIA drivers.

**Comment by effofxprime** [comment]
> > > What does the mining console say when the crash happens?
> > 
> > 
> > "Miner crashed, watchdog restarting."
> > I didn't see an indication as to why, like GPU failed or anything like that. It was fast and I couldn't screenshot before it closed and restarted.
> > I only saw this for the first time today because of how automated it runs. I use awesome miner. But I turned on the watchdog for it to restart the process without hanging the rig and requiring a restart.
> > I didn't see anything in the window about a why. However, I have turned on logging. Let me check the log, screenshot and get back with that information as well.
> 
> Did you try disabling all the antivirus features? Try changing UAC to allow all. Reinstall NVIDIA drivers.

No antivirus enabled.  It took me a bit to figure out what was really going on.  I think it was a 2 part issue.

Windows 10 updates.  I thought I had disabled that shit.  Now I did so via group policy and recommend that others do the same.
Probably was a P2 issue (partly) from updates that reset that.  I am happy that gminer is able to force it via a CLI switch.  Thank you for adding that.

Secondly I think there were some hardware issues.  I have been running a 8GPU rig for 8 months now and it only started doing this recently.  I narrowed down the cards that I was pretty sure having issues.  Reapplied thermal paste on those cards, reinstalled and now I am running good again.

Hardware issues are interesting.  Prior, I couldn't use the restart mining feature on Awesomeminer without it causing the machine to freeze.  Since I did the above things, this morning I am able to restart the process without the machine freezing.  Now I will just wait to see if after some time if it will continue to restart by itself or if that's now been resolved.
Closing this as I believe the issue is resolved and not a gminer issue.  I am wondering if any attempt can be made that would allow gminer to throw a log/error something that would indicate hardware failure?
